tests/src/core/testqgsexpression.cpp

```diff
@@ -2229,14 +2229,14 @@ class TestQgsExpression: public QObject
       // Two calls with the same seed always return the same number
       QgsExpression exp8( QStringLiteral( "randf(seed:=1)" ) );
       QVariant v8 = exp8.evaluate();
-      QCOMPARE( v8.toFloat() == 0.13387664401253274, true );
+      QCOMPARE( v8.toDouble() == 0.13387664401253274, true );
```

Comment:
QCOMPARE can be used to actually compare values, so in general it's better to do `QCOMPARE( val1, val2 )` than `QCOMPARE(val1==val2, true)`.
But since double comparison is special you are better off with `QVERIFY( qgsDoubleNear( v8.toDouble(), value ) )` in this particular case

Reply:
Thanks Matthias :-) I thought I'd never get the test pass...